vortex-buffer/src/string.rs

```diff
@@ -64,10 +65,14 @@ impl From<&str> for BufferString {
 }
 
 impl TryFrom<ByteBuffer> for BufferString {
-    type Error = Utf8Error;
+    type Error = VortexError;
 
     fn try_from(value: ByteBuffer) -> Result<Self, Self::Error> {
-        let _ = std::str::from_utf8(value.as_ref())?;
+        let _ = simdutf8::basic::from_utf8(value.as_ref()).map_err(|_| {
+            #[allow(clippy::unwrap_used)]
+            let err = simdutf8::compat::from_utf8(value.as_ref()).unwrap_err();
```

Comment:
wat? doesn't it pass in the err? you're already in map_err from the same call

Reply:
Basic is basic. The error doesn’t have any information to make it a zero sized type

Reply:
is compat that much slower? seems weird to run validation twice